### ISSUE TITLE
Topic Metrics not persistent after restart EMQX instance

### ISSUE THREAD

**Issue by N0tronic** [emqx/emqx#8264]
### What happened?

I use the Rest Interface to getting information about a few topics. I create the topic metrics with the EMQX Dashboard (Analysis/Topic metrics). But after restarting the EMQX Instances the topic metrics no more there. I persist all preferences like this:

- vol-broker-data:/opt/emqx/data
- vol-broker-etc:/opt/emqx/etc
- vol-broker-log:/opt/emqx/log

All preferences are there after restarting EMQX instances. But not the topic metrics.

### What did you expect to happen?

That topic metrics are still there after EMQX instances restart.

### How can we reproduce it (as minimally and precisely as possible)?

Create a topic metric over the EMQX Dashboard (Analysis/Topic metrics) and then restart the docker container. Volumes are mounted:

- vol-broker-data:/opt/emqx/data
- vol-broker-etc:/opt/emqx/etc
- vol-broker-log:/opt/emqx/log

### Anything else we need to know?

All other EMQX preferences are persistent after restart the container.

### EMQX version

4.3.15

### OS version

EMQX Docker Container

### Log files

<details>

</details>

**Comment by N0tronic** [comment]
Interesting, I monitor some topics in grafana over the rest interface /api/v4/topic-metrics
@lafirest Why can´t make data that I can get over a rest interface not persistent?

So I have to configure each topic metric again after a restart???

**Comment by lafirest** [comment]
> 

No, the data of the metric is not persistent, but the metric you set should be. I think `configure each topic metric again after a restart` is a bug

**Comment by zmstone** [comment]
v2 topic metrics now supports wildcard match, and stores the named collections in database.
https://github.com/emqx/emqx/pull/17607
